### PR TITLE
Destination column types redux

### PR DESCRIPTION
**Pull request by wemrysi**
[ch7196]
[ch9155]
[ch8624]

### CODE REVIEW

**Comment by wemrysi** [comment]
@jsantos17 Yes, we'll want to wait to merge this since it breaks things. I have all the SDBE changes completed and a snapshot released so that integration can be implemented, I'll remove the :stop_sign: here once it is ready to merge.

**Comment by alissapajer** [comment]
@wemrysi what's the status of this PR?

**Comment by wemrysi** [comment]
@alissapajer awaiting frontend support which @cryogenian is currently working on.

File: api/src/main/scala/quasar/api/destination/LegacyDestination.scala [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+import cats.data.NonEmptyList
+
+import monocle.Prism
+
+import quasar.api.table.ColumnType
+
+import scala.Int
+
+trait LegacyDestination[F[_]] extends UnparameterizedDestination[F] {
```
Comment (alissapajer) [inline api/src/main/scala/quasar/api/destination/LegacyDestination.scala]: @wemrysi why is a legacy destination required? Could we migrate existing destinations to use the new destination definition?

Comment (wemrysi) [reply api/src/main/scala/quasar/api/destination/LegacyDestination.scala]: @alissapajer we can, but having this allows us to decouple the processes, we can land destination types, having all destinations use `LegacyDestination` or `UntypedDestination` as appropriate, and then go back and add the custom types and handling to the destinations where it makes sense.

**Pull request merged**